File: android/PetSitterISI/app/src/main/java/com/example/petsitterisi/entitees/Promotion.java
package com.example.petsitterisi.entitees;

import java.util.Date;

public class Promotion {

    private int id;
    private String codePromotion;
    private Date dateFin;
    private int pourcentage;

    public Promotion(int id, String codePromotion, Date dateFin, int pourcentage) {
        this.id = id;
        this.codePromotion = codePromotion;
        this.dateFin = dateFin;
        this.pourcentage = pourcentage;
    }

    public int getId() {
        return id;
    }

    public String getCodePromotion() {
        return codePromotion;
    }

    public Date getDateFin() {
        return dateFin;
    }

    public int getPourcentage() {
        return pourcentage;
    }
}

File: backend/src/models/promotion.js
class Promotion {
    constructor(id, code_promotion, pourcentage, date_fin)
    {
        this._id = id
        this._code_promotion = code_promotion
        this._pourcentage = pourcentage
        this._date_fin = date_fin
    }

    /* ######################################### GETTER #########################################*/

    get id(){
        return this._id
    }
    get code_promotion(){
        return this._code_promotion
    }
    get pourcentage(){
        return this._pourcentage
    }
    get date_fin(){
        return this._date_fin
    }

    /* ######################################### GETTER #########################################*/

    set id(id){
        this._id = id
    }
    set code_promotion(code_promotion){
        this._code_promotion = code_promotion
    }
    set pourcentage(pourcentage){
        this._pourcentage = pourcentage
    }
    set date_fin(date_fin){
        this._date_fin = date_fin
    }
}

module.exports = Promotion